src/backend/access/appendonly/aomd.c

```diff
@@ -250,6 +251,9 @@ mdunlink_ao_perFile(const int segno, void *ctx)
 	char *segPathSuffixPosition = unlinkFiles->segpathSuffixPosition;
 
 	sprintf(segPathSuffixPosition, ".%u", segno);
+
+	do_truncate(segPath);
```

Comment:
Why are the conditions after calling the do_truncate and unlink functions different in the mdunlink_ao_perFile and mdunlink functions? In the mdunlink_ao_perFile function, the result of the do_truncate function is not checked at all, the unlink function is called unconditionally, and its result is compared only with 0. Whereas in the mdunlink function, the result of the do_truncate function is compared with 0, and the unlink function is called only under certain conditions.

Reply:
Changed to be similar to segment files handling in mdunlinkfork